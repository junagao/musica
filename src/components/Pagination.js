import React from 'react';
import PropTypes from 'prop-types';

const Pagination = ({ albumsPerPage, totalAlbums, onChangePage }) => {
  const pageNumbers = [];

  for (let i = 1; i <= Math.ceil(totalAlbums / albumsPerPage); i += 1) {
    pageNumbers.push(i);
  }

  return (
    <div>
      <nav>
        <ul className="pagination">
          {pageNumbers.map((number) => (
            <li key={number} className="page-item">
              <a
                onClick={() => onChangePage(number)}
                href={`#${number}`}
                className="page-link"
              >
                {number}
              </a>
            </li>
          ))}
        </ul>
      </nav>
    </div>
  );
};

Pagination.propTypes = {
  albumsPerPage: PropTypes.number.isRequired,
  totalAlbums: PropTypes.number.isRequired,
  onChangePage: PropTypes.func.isRequired,
};

export default Pagination;
