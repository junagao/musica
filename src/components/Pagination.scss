.pagination {
  display: flex;
  margin-bottom: 2.5rem;
}

.arrow {
  margin-top: 0.45rem;
}
