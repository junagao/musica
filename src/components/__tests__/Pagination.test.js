import React from 'react';
import { shallow } from 'enzyme';

import Pagination from 'components/Pagination';

describe('Pagination', () => {
  let wrapper;
  const albummsPerPageMock = 5;
  const totalAlbumsMock = 40;
  const onChangePageMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Pagination
        albumsPerPage={albummsPerPageMock}
        totalAlbums={totalAlbumsMock}
        onChangePage={onChangePageMock}
      />,
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one div element', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render one nav element', () => {
    expect(wrapper.find('nav').length).toEqual(1);
  });

  it('should render one ul element', () => {
    expect(wrapper.find('ul').length).toEqual(1);
  });

  it('should render 8 li element with className `page-item`', () => {
    expect(wrapper.find('.page-item').every('.page-item')).toBeTruthy();
  });
});
