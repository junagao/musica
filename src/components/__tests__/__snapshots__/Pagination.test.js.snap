// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly 1`] = `
<div>
  <nav>
    <ul
      className="pagination"
    >
      <li
        className="page-item"
        key="1"
      >
        <a
          className="page-link"
          href="#1"
          onClick={[Function]}
        >
          1
        </a>
      </li>
      <li
        className="page-item"
        key="2"
      >
        <a
          className="page-link"
          href="#2"
          onClick={[Function]}
        >
          2
        </a>
      </li>
      <li
        className="page-item"
        key="3"
      >
        <a
          className="page-link"
          href="#3"
          onClick={[Function]}
        >
          3
        </a>
      </li>
      <li
        className="page-item"
        key="4"
      >
        <a
          className="page-link"
          href="#4"
          onClick={[Function]}
        >
          4
        </a>
      </li>
      <li
        className="page-item"
        key="5"
      >
        <a
          className="page-link"
          href="#5"
          onClick={[Function]}
        >
          5
        </a>
      </li>
      <li
        className="page-item"
        key="6"
      >
        <a
          className="page-link"
          href="#6"
          onClick={[Function]}
        >
          6
        </a>
      </li>
      <li
        className="page-item"
        key="7"
      >
        <a
          className="page-link"
          href="#7"
          onClick={[Function]}
        >
          7
        </a>
      </li>
      <li
        className="page-item"
        key="8"
      >
        <a
          className="page-link"
          href="#8"
          onClick={[Function]}
        >
          8
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
